### PR TITLE
GraphQLHttpMiddleware returns 400 when query is missing

### LINKED ISSUE #583: GraphQL endpoint throwing HTTP 500 on HTTP GET
## Description

The graphQL server throws an InvalidOperationException("Cannot execute request if no query is specified") when HTTP GET is requested on graphql endpoint.

This is a very common scenario (people opening the graphql endpoint in browser). It is causing an internal server error. It would be better if the GraphQL framework could handle this situation. Instead the error is propagated to ASP.NET middleware.

### Steps to reproduce

Open url `http://{server}/graphql` in browser which points to the graphql endpoint.

### Expected result

Bad http request (400)

### Actual result

Unhandled exception is thrown.
Returned Http Response 500.

### PR DESCRIPTION
**Pull request by sharppanda**
Fixes #583

### CODE REVIEW

File: src/Transports.AspNetCore/GraphQLHttpMiddleware.cs [hunk 1]
```diff
@@ -120,6 +120,15 @@ public async Task InvokeAsync(HttpContext context)
                     Extensions = urlGQLRequest.Extensions ?? bodyGQLRequest?.Extensions,
                     OperationName = urlGQLRequest.OperationName ?? bodyGQLRequest?.OperationName
                 };
+
+                if (string.IsNullOrWhiteSpace(urlGQLRequest.Query))
```
Comment (Shane32) [inline src/Transports.AspNetCore/GraphQLHttpMiddleware.cs:124]:
```suggestion
                if (string.IsNullOrWhiteSpace(gqlRequest.Query))
```

**Review by Shane32 — CHANGES_REQUESTED** [review]

**Review by Shane32 — APPROVED** [review]
Looks good to me but @sungam3r needs to approve it.

**Comment by sharppanda** [comment]
Do you have an ETA when this could be reviewed by @sungam3r ?

**Comment by Shane32** [comment]
@sungam3r bump

**Comment by sungam3r** [comment]
@sharppanda Thanks for fixing this. 

> Do you have an ETA when this could be reviewed by @sungam3r ?

I had a long break from OSS this year so...just reviewed! LGTM

**Pull request merged**